src/containers/collaborations/components/AddDataSetToCollaborationModal.js

```diff
@@ -0,0 +1,193 @@
+  isVisible,
+  onClose,
+  organizationIds,
+  ...rest
+} :{
+  collaborationId :UUID;
+  existingDataSets :List;
+  isVisible :boolean;
+  onClose :() => void;
+  organizationIds :List<UUID>;
+}) => {
+  const dispatch = useDispatch();
+  const [modalTitle, setModalTitle] = useState(<Typography variant="h3">Select Organization</Typography>);
+  const [step, stepBack, stepNext, setStep] = useStepState(3);
+  const [selectedOrganization, setSelectedOrganization] = useState('');
+  const [selectedDataSets, setSelectedDataSets] = useState(List());
+  const [selectedDataSetTitles, setSelectedDataSetTitles] = useState(List());
+  const addDataSetToCollaborationRS :?RequestState = useRequestState([COLLABORATIONS, ADD_DATA_SETS_TO_COLLABORATION]);
+
+  const organizations :Map<UUID, Organization> = useSelector(selectOrganizations());
+
+  const pending = isPending(addDataSetToCollaborationRS);
+  const standby = isStandby(addDataSetToCollaborationRS);
+  const success = isSuccess(addDataSetToCollaborationRS);
+
+  const orgOptions = organizationIds.filter((orgId) => organizations.has(orgId))
+    .map((orgId) => {
+      const organization = organizations.get(orgId);
+      const label = organization.title;
+      const value = organization.id;
+      return { label, value };
+    }).toJS();
+
+  useEffect(() => {
+    if (step === 0) {
+      setModalTitle(<Typography variant="h3">Select Organization</Typography>);
+    }
+    else if (step === 1) {
+      setModalTitle(<Typography variant="h3">Add Data Sets</Typography>);
+    }
+    else if (standby && step === 2) {
+      setModalTitle(<Typography variant="h3">Add these data sets?</Typography>);
+    }
+    else if (success && step === 2) {
+      setModalTitle(<Typography variant="h3">Data Sets Added!</Typography>);
+    }
+  }, [setModalTitle, standby, step, success]);
+
+  const handleOrgSelect = (organizationOption :ReactSelectOption<Object>) => {
+    setSelectedOrganization(organizationOption.value);
+  };
+
+  const closeModal = () => {
+    onClose();
+    setStep(0);
+  };
+
+  const handleOnClickPrimary = () => {
+    dispatch(
+      addDataSetsToCollaboration({
+        collaborationId,
+        dataSetIdsByOrgId: Map({
+          [selectedOrganization]: selectedDataSets
+        })
+      })
+    );
+  };
+
+  return (
+    <Modal
+      // eslint-disable-next-line
+        {...rest}
+        isVisible={isVisible}
+        onClose={closeModal}
+        textTitle={modalTitle}>
+      {
+        step === 0 && (
+          <>
+            <ModalBody>
+              <StackGrid gap={16}>
+                <Typography gutterBottom htmlFor="selectOrganization">
+                  Which organization would you like to add a dataset from?
+                </Typography>
+                <Select
+                    id="select organization"
+                    menuPortalTarget={document.body}
+                    styles={{ menuPortal: (provided) => ({ ...provided, zIndex: 9999 }) }}
```

Comment:
I feel like we shouldn't be scrolling within the modal to see the contents of the select component. The Menu should be indexed above the component it's in and should not be bound by the size of that component.

Reply:
oh i see. i suppose they do the same thing in the end. with viewportScrolling. the select menu is indexed over the modal and can escape. you scroll the entire window instead of the contents of the modal which i find to be a little more helpful esp if your window size is small. it also scrolls the window to automatically center the opened select. i don't really mind which way we do it just thought i'd share another way to do it. should discuss with chris which way is preferred.

![image](https://i.gyazo.com/58a08b87ceafead537c620413acd576f.gif)